### PR TITLE
Multiple search queries

### LINKED ISSUE #82: Be able to search for multiple items at once
Right now, the Search API only considers the first term we search for.
We need to be able to search on multiple queries at once, then either intersect (for) AND or union (for OR) the results.

### PR DESCRIPTION
**Pull request by Winwardo**
Closes #82 

@VasilyShelkov ready for comment + merge

### CODE REVIEW

File: src/server/tweetFinder.js [hunk 1]
```diff
@@ -10,40 +10,74 @@ import { searchAndSaveFromTwitter } from './twitterSearch';
...
+    data: {
+      count: results.length,
+      records: results,
+    },
+  }
 );
 
 const searchDatabase = (query, alreadyAttemptedRefresh = false) => {
-  const tweetSelection = 'SELECT *, in(\'TWEETED\').id AS authorId, in(\'TWEETED\').name AS authorName, in(\'TWEETED\').handle AS authorHandle FROM tweet WHERE content LUCENE :query ORDER BY date DESC UNWIND authorId, authorName, authorHandle LIMIT 300';
+  const tweetSelection = 'SELECT *, in(\'TWEETED\').id AS authorId, in(\'TWEETED\').name AS authorName, in(\'TWEETED\').handle AS authorHandle FROM tweet WHERE content LUCENE :query ORDER BY date DESC UNWIND authorId, authorName, authorHandle LIMIT :limit';
```
Comment (Winwardo) [inline src/server/tweetFinder.js]: Codacy Issue found: [Line is too long.](https://www.codacy.com/app/winwardo/solid-octo-disco/file/2311644437/issues/source?bid=3148782&fileBranchId=3199135#l77)

Comment (Winwardo) [inline src/server/tweetFinder.js]: Codacy Issue found: [Line is too long.](https://www.codacy.com/app/winwardo/solid-octo-disco/file/2313672725/issues/source?bid=3148782&fileBranchId=3199135#l77)

**Comment by VasilyShelkov** [comment]
fine with the code, a couple of questions on the overalll design before merging:

1) What led you to decide that 'splattingTogether' post searching is the way to OR. What's the reasoning behind doing this instead of 'oring' and in the future potentitally 'anding' on the query level with the twitter API and/or our database query.
image

2) If we decide to do ANDing the query, how would we 'splatTogether' them approximately ? since I'm not sure how we would easily (Again where the alternative would be doing the 'and' query operator from twitter/some way from our database)

**Comment by Winwardo** [comment]
1) I don't feel it makes sense to do specifically Twitter ORing / ANDing - although we could add an extra query on top of each specific query. I say this as the way we search our own database will fundamentally return different results to what searching Twitter would directly, as we're doing fuzzy searching and have access to a different subset of Tweets.
Secondly, we COULD build up a complex OR query to send to OrientDB, however I believe this would be an extra level of complexity to implement and would be less stable. Lots of different queries, then OR/ANDing afterwards is very simple to implement and test with little chance for failure.

2) Same way we OR, but by implementing intersectTweets instead of unionTweets - here it is in Python https://stackoverflow.com/questions/642763/python-intersection-of-two-lists

File: src/server/tweetFinder.js (hunk 1 above)
Comment (Winwardo) [inline src/server/tweetFinder.js:77]: Codacy Issue found: [Line is too long.](https://www.codacy.com/app/winwardo/solid-octo-disco/file/2316860133/issues/source?bid=3148782&fileBranchId=3199135#l77)

**Pull request merged**